scio-repl jobs when using DataflowRunner throw exception

```
java.lang.RuntimeException: org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.U
ncheckedExecutionException: java.lang.ClassCastException: cannot assign instance of java.lang.invoke.SerializedLambda to field com.spotify.scio.util.Functions$$anon$3.g of type scala.Function1 in instance of com
.spotify.scio.util.Functions$$anon$3
        at org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:194)
        at org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:165)
        at org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:63)
        at org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:50)
        at org.apache.beam.runners.dataflow.worker.graph.Networks.replaceDirectedNetworkNodes(Networks.java:87)
        at org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.create(IntrinsicMapTaskExecutorFactory.java:125)
        at org.apache.beam.runners.dataflow.worker.BatchDataflowWorker.doWork(BatchDataflowWorker.java:358)
        at org.apache.beam.runners.dataflow.worker.BatchDataflowWorker.getAndPerformWork(BatchDataflowWorker.java:311)
        at org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.doWork(DataflowBatchWorkerHarness.java:140)
        at org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.call(DataflowBatchWorkerHarness.java:120)
        at org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.call(DataflowBatchWorkerHarness.java:107)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.UncheckedExecutionException: java.lang.ClassCastException: cannot assign instance of java.lang.invoke.SerializedLambda to field
 com.spotify.scio.util.Functions$$anon$3.g of type scala.Function1 in instance of com.spotify.scio.util.Functions$$anon$3
        at org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2050)
        at org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache.get(LocalCache.java:3952)
        at org.apache.beam.vendor.guava.v26_0_jre.com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4871)
        at org.apache.beam.runners.dataflow.worker.UserParDoFnFactory.create(UserParDoFnFactory.java:91)
        at org.apache.beam.runners.dataflow.worker.DefaultParDoFnFactory.create(DefaultParDoFnFactory.java:75)
        at org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.createParDoOperation(IntrinsicMapTaskExecutorFactory.java:264)
        at org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.access$000(IntrinsicMapTaskExecutorFactory.java:86)
        at org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$1.typedApply(IntrinsicMapTaskExecutorFactory.java:183)
        ... 14 more
```

Classpath is incomplete.

--- a/scio-repl/src/main/scala-2.12/com/spotify/scio/repl/ScioILoop.scala
+++ b/scio-repl/src/main/scala-2.12/com/spotify/scio/repl/ScioILoop.scala
@@ -28,6 +28,7 @@ import org.apache.commons.text.StringEscapeUtils
 import scala.tools.nsc.{CompilerCommand, Settings}
 import scala.tools.nsc.interpreter.Results
 import scala.tools.nsc.interpreter.ILoop
+import scala.reflect.io.AbstractFile
 
 /**
  * ScioILoop - core of Scio REPL.
@@ -254,4 +255,6 @@ class ScioILoop(command: CompilerCommand, scioClassLoader: ScioReplClassLoader, 
   }
 
   def run(settings: Settings): Boolean = process(settings)
+
+  def outputDir: AbstractFile = intp.replOutput.dir
 }

--- a/scio-repl/src/main/scala-2.12/com/spotify/scio/repl/compat/ILoopClassLoader.scala
+++ b/scio-repl/src/main/scala-2.12/com/spotify/scio/repl/compat/ILoopClassLoader.scala
@@ -1,11 +1,12 @@
 package com.spotify.scio.repl.compat
 
 import scala.tools.nsc.interpreter.ILoop
+import com.spotify.scio.repl.ScioILoop
 
 private[repl] trait ILoopClassLoader {
 
-  protected var scioREPL: ILoop = _
+  protected var scioREPL: ScioILoop = _
 
-  def setRepl(repl: ILoop): Unit = scioREPL = repl
+  def setRepl(repl: ScioILoop): Unit = scioREPL = repl
 
 }

--- a/scio-repl/src/main/scala-2.13/com/spotify/scio/repl/compat/ILoopClassLoader.scala
+++ b/scio-repl/src/main/scala-2.13/com/spotify/scio/repl/compat/ILoopClassLoader.scala
@@ -1,11 +1,12 @@
 package com.spotify.scio.repl.compat
 
 import scala.tools.nsc.interpreter.shell.ILoop
+import com.spotify.scio.repl.ScioILoop
 
 private[repl] trait ILoopClassLoader {
 
-  protected var scioREPL: ILoop = _
+  protected var scioREPL: ScioILoop = _
 
-  def setRepl(repl: ILoop): Unit = scioREPL = repl
+  def setRepl(repl: ScioILoop): Unit = scioREPL = repl
 
 }

--- a/scio-repl/src/main/scala/com/spotify/scio/repl/ScioReplClassLoader.scala
+++ b/scio-repl/src/main/scala/com/spotify/scio/repl/ScioReplClassLoader.scala
@@ -27,6 +27,7 @@ import com.spotify.scio.repl.compat.ILoopClassLoader
 import org.slf4j.LoggerFactory
 
 import scala.tools.nsc.io._
+import scala.collection.AbstractIterable
 
 object ScioReplClassLoader {
   private val Logger = LoggerFactory.getLogger(this.getClass)
@@ -109,18 +110,10 @@ class ScioReplClassLoader(urls: Array[URL], parent: ClassLoader)
    */
   private[scio] def createReplCodeJar: String = {
     require(scioREPL != null, "scioREPL can't be null - set it first!")
-    import scala.tools.nsc.interpreter.ReplOutput
-
-    val virtualDirectory = new ReplOutput(scioREPL.settings.Yreploutdir).dir
-
     val tempJar = new File(nextReplJarDir, replJarName)
-
     // Generate next repl jar dir
     nextReplJarDir = genNextReplCodeJarDir
-
-    val jarFile =
-      createJar(virtualDirectory.asInstanceOf[VirtualDirectory], tempJar)
-    jarFile.getPath
+    createJar(scioREPL.outputDir, tempJar).getPath()
   }
 
   /**
@@ -128,13 +121,10 @@ class ScioReplClassLoader(urls: Array[URL], parent: ClassLoader)
    *
    * @param virtualDirectory containing classes that should be added to the jar
    */
-  private def createJar(virtualDirectory: VirtualDirectory, jarFile: File): File = {
+  private def createJar(dir: AbstractIterable[AbstractFile], jarFile: File): File = {
     val jarStream = new JarOutputStream(new FileOutputStream(jarFile))
-    try {
-      addVirtualDirectoryToJar(virtualDirectory, "", jarStream)
-    } finally {
-      jarStream.close()
-    }
+    try { addVirtualDirectoryToJar(dir, "", jarStream) }
+    finally { jarStream.close() }
 
     jarFile
   }
@@ -148,24 +138,21 @@ class ScioReplClassLoader(urls: Array[URL], parent: ClassLoader)
    * @param jarStream for writing the jar file
    */
   private def addVirtualDirectoryToJar(
-    dir: VirtualDirectory,
+    dir: AbstractIterable[AbstractFile],
     entryPath: String,
     jarStream: JarOutputStream
-  ): Unit =
-    dir.foreach { file =>
-      if (file.isDirectory) {
-        // Recursively descend into subdirectories, adjusting the package name as we do.
-        val dirPath = entryPath + file.name + "/"
-        val entry: JarEntry = new JarEntry(dirPath)
-        jarStream.putNextEntry(entry)
-        jarStream.closeEntry()
-        addVirtualDirectoryToJar(file.asInstanceOf[VirtualDirectory], dirPath, jarStream)
-      } else if (file.hasExtension("class")) {
-        // Add class files as an entry in the jar file and write the class to the jar.
-        val entry: JarEntry = new JarEntry(entryPath + file.name)
-        jarStream.putNextEntry(entry)
-        jarStream.write(file.toByteArray)
-        jarStream.closeEntry()
-      }
+  ): Unit = dir.foreach { file =>
+    if (file.isDirectory) {
+      // Recursively descend into subdirectories, adjusting the package name as we do.
+      val dirPath = entryPath + file.name + "/"
+      jarStream.putNextEntry(new JarEntry(dirPath))
+      jarStream.closeEntry()
+      addVirtualDirectoryToJar(file, dirPath, jarStream)
+    } else if (file.hasExtension("class")) {
+      // Add class files as an entry in the jar file and write the class to the jar.
+      jarStream.putNextEntry(new JarEntry(entryPath + file.name))
+      jarStream.write(file.toByteArray)
+      jarStream.closeEntry()
     }
+  }
 }

--- a/scio-repl/src/main/scala/com/spotify/scio/repl/ScioReplClassLoader.scala
+++ b/scio-repl/src/main/scala/com/spotify/scio/repl/ScioReplClassLoader.scala
@@ -121,7 +121,7 @@ class ScioReplClassLoader(urls: Array[URL], parent: ClassLoader)
    *
    * @param virtualDirectory containing classes that should be added to the jar
    */
-  private def createJar(dir: AbstractIterable[AbstractFile], jarFile: File): File = {
+  private def createJar(dir: Iterable[AbstractFile], jarFile: File): File = {
     val jarStream = new JarOutputStream(new FileOutputStream(jarFile))
     try { addVirtualDirectoryToJar(dir, "", jarStream) }
     finally { jarStream.close() }
@@ -138,7 +138,7 @@ class ScioReplClassLoader(urls: Array[URL], parent: ClassLoader)
    * @param jarStream for writing the jar file
    */
   private def addVirtualDirectoryToJar(
-    dir: AbstractIterable[AbstractFile],
+    dir: Iterable[AbstractFile],
     entryPath: String,
     jarStream: JarOutputStream
   ): Unit = dir.foreach { file =>

